### PR TITLE
Generate R docs

### PR DESCRIPTION
**Pull request by abal5**
#602 

To add the generation of R documentation for this project.
Using a small container from dockerhub that has R 3.1 on it, install the packages needed, build the docs, archive the doc files.

### CODE REVIEW

**Comment by abal5** [comment]
I rather have this build separately as well so I'll look into making this it's own jenkinsfile.

**Comment by navdeep-G** [comment]
Stage looks good. I agree it should be built separately.

File: Jenkinsfile.utils2 [hunk 1]
```diff
@@ -199,6 +199,23 @@ pipeline {
...
+            steps {
+                sh """
+                    R CMD INSTALL usethis
+                    R CMD INSTALL devtools
+                    R CMD INSTALL  pandoc
+                    wget https://github.com/r-lib/pkgdown/archive/v1.0.0.tar.gz
+                    R CMD INSTALL pkgdown-1.0.0.tar.gz
+                    R && `pkgdown::build_site(examples=FALSE)` && q(save="no")
+                   """
+                archiveArtifacts artifacts: 'src/interface_r/docs/*.html'
```
Comment (terrytangyuan) [inline Jenkinsfile.utils2]: This doesn’t include images if there’s any? Also there’s a index.html in a directory above that you might need as well

**Comment by abal5** [comment]
Ok so the R documentation is now being generated and being uploaded to S3, the only thing left is we need to decide what kind of url/folder path we would like and also how versioning should be handled.

**Pull request closed without merging**